### DISCUSSION TITLE
Setup Assistant - Configure Video Sources - Channels scan

### DISCUSSION THREAD

**Discussion by bradut** [GerryDazoo/Slinger#197] [Q&A] [answered]
Thank you for this great implementation Gerry !

I've got it working with my digital cable TV in Canada and soon will try to connect to a couple of SlingBoxes overseas.
Those boxes receive analogue signal from antena coaxial cable (ANT IN) .

Questions:
- Is it possible to run a "Channel scan" with the current implementation of Slinger?
- Is it possible to provide Slinger with a list of frequencies for each of those boxes? 
 
 Regards,
 Bradut
  
![SlingBox setup 02 - Configure Video Sources - 03 Channel scaned_MTV](https://user-images.githubusercontent.com/2496596/198862064-bf9badb3-3fc0-4d4f-aedc-7684341f6136.jpg)

**Comment by RandomMan719** [comment]
I am wondering the same thing.  I just found out that Slingplayer 1.5.1 won't scan for Digital Channels, only Analog.  Slingplayer 2.0 is the first version that allows for Digital Channel Scan, and it requires logging into your account.
![image](https://user-images.githubusercontent.com/116921195/198872063-bdc56e3a-afa3-4afc-9630-b2cedd6aa0ab.jpeg)

**Reply by BobC1954** [reply]
I use Slingplayer 1.5.1.342 occasionally with a ProHD. It does allow tuning of digital channels. Have you tried clicking next on the screen you show? I retuned one of my boxes a few days ago using this version and it worked fine for digital. However, 1.5.1.342 has become quite flakey  and does not always successfully stream the box, just showing "pending". I don't know why it does this.

**Reply by RandomMan719** [reply]
My version is slightly older.  Do you have a link to that version?  
![Version](https://user-images.githubusercontent.com/116921195/198892931-74935c50-22a0-41d2-ac72-fbc6e9f961a9.PNG)

**Reply by BobC1954** [reply]
v1.5.1.something is on fyxm.net But I don't know what the "something" (last version) is. I have the installer for .342 somewhere, but might take some time to search many hard disks and it is about 80megs so a bit big to upload.

**Comment by GerryDazoo** [comment, marked as answer]
The short answer is no and no.

**Reply by bradut** [reply]
Maybe I should try sniffing the traffic between the SlingBox and the Router to see how analogue channel-change messages look like. 
What do you think?

**Reply by RandomMan719** [reply]
Yes please

**Reply by GerryDazoo** [reply]
Can we be clear here. Are we talking about channel scanning or channel changing? The next version coming soon supports (digital?) channel changing on the built in ProHD tuner. 
How does one select which tuner to use? How do you tell it to tune to a analog channel now? Yes a packet capture between the Desktop and slingbox would be helpful. 
Are you running Python or the exe? I can get you a test version.

**Reply by bradut** [reply]
Hi Gerry.

- I am interested in channel changing
  - For now I can skip the channel scanning as I wrote down their numbers and frequencies. 
  - Unfortunately I can only use the analog signal for those remote boxes.

- How to select which tuner to use? 
  - by using csv files.
    - having the same names as the tuners declared in the `unified_config.ini`, such as `Solo.csv`, `500.csv`, `M1.csv`
    - fields [`IsActive`, `ChannelNo`, `Frequency`] 
  
- How do you tell it to tune to a analog channel now
  - the server will know that the channel is analog by checking if the corresponding csv file exists.
  - I don't know what info to be sent to SlingBox :(.  I am trying to investigate with Wireshark, but cannot get analogue signal from my cable provider (TELUS) here in Canada. Unfortunately I will be able to reach out to the remote boxes only next year.
  
 - I am running the exe. Planning to learn Python... 
 
 Thank you !

**Comment by RandomMan719** [comment]
I guess you will need to hookup a converter box, to another input, and get the remote working to control the converter box and scan for channels on it.  I do that with my Slingbox Solos, I will do the same with the ProHD, as well.  My converter boxes only put out SD, which is fine for my use.

**Reply by bradut** [reply]
Interesting workaround.
Could you please let me know the type/model of the converter box?

**Reply by sp222222** [reply]
do search on ebay for hdmi TO component. i had to order a backup this week  “tv box vhs hdmi to component YPBPR male converter 720” 21$

**Comment by sp222222** [comment]
I pulled my old ProHD out earlier this month and was able to scan (50+ channels) for ATSC digital channels using the TUNER input. I also have as a fallback a converter box using component converter taking the HDMI signal from the converter box and feeding the IN on the solo component (this box has USB so I can record). I think I used the slingplayer desktop (windows) to set port, IP and inputs up. I hope to never have to scan for channels again but who knows.

**Reply by RandomMan719** [reply]
What version of SlingPlayer did you use, I can scan for digital channels with SlingPlayer 2.0 and above, but I have to login to my account to scan.  Version 1.51, I don't have to login my account, but I am only getting the option to scan for Analog channels.

**Reply by sp222222** [reply]
1.5.1.338 - I have it on my google drive - send me your email and I'll let you snag a copy.

**Reply by proudx2** [reply]
I posted this to the discord.  Option for scanning channels post nov 9 is using the 1.5 slingplayer builds to setup and configure inputs and needing to scan channels. 

I found an option under main of sbcore.ini called antennascan=1 for selecting antenna scanning as an option. It might be listed as 0 in your core and needs to be on otherwise you eill not get an option to use antenna and only cable connected option. Also might need to do an advanced scan or detailed scan in addition to pickup all channels.

For viewing you would obviously need slingplayer 2 or gerrys new build to see them.

also on the 1.51 builds 338 is the latest us build I think And 343 has a -eu after it in the installer name. Perhaps eurropean union version?  It has an additional fileunder sbav called sbav-ju thst I needed to add to my us 338!build sbav directory to be able to configure remote ir inputs  on my tested hdpro. 

I know it doesn’t say digital scan. But in theory an advanced scan should see those same shared uhf/vhf frequencies as retired analog ntsc

**Reply by Mike7766** [reply]
Is there step by step instructions on how to setup slingbox with slinger

**Reply by RandomMan719** [reply]
Read the Readme, the Release Notes, and the Networking  Guide.  It's pretty easy.

**Reply by Mike7766** [reply]
I tried if you can help would be appreciated

**Reply by sp222222** [reply]
it doesn't scan digital and I can no longer test my ProHD because it bricked yesterday.

Mike7766 - set up block rules on your router where the slingbox sits, else you risk bricking the hardware.

**Reply by RandomMan719** [reply]
Can you still ping bricked ProHD's?

Also can you connect to a bricked one with Slingplayer 1.5.1?  There is a place to update firmware in properties, I believe

**Reply by sp222222** [reply]
sure, but how much will that do? it's bricked until firmware is flashed back to normal.

**Reply by sp222222** [reply]
the firmware button along with every other button on 1.5 is blocked out b/c of error connecting to the device. 
admin pass no longer works on mine either. sad thing is , I had it working again, but hit the hard reset once to many times thinking it wasn't working.

**Reply by RandomMan719** [reply]
So you can ping it?  Have you pinged it?  I just wanna know because I have some ProHDs at remote sites that I can ping, but they seem like they are bricked.

**Reply by sp222222** [reply]
it pings (with the new IP I gave it) and if you cannot view the stream, it's probably bricked.

**Reply by RandomMan719** [reply]
Yeah, I blocked them all from the internet yesterday morning.  I am thinking they are bricked, won't know until I can go to each site.  Hopefully we can find a way to flash the firmware, and get them functioning again.

I was able to setup a Solo (hasn't been powered up for months and I blocked it from the internet) a couple hours ago, with Slingplayer 2.0.4 and it even logged into my account and I could see all my slingboxes in the directory.  I just renamed the solo, set the password, and configured a composite video source and add a custom remote for a Zenith Converter box.

**Reply by Mike7766** [reply]
Can someone help me with setup on sling

**Reply by RandomMan719** [reply]
Mike7766, what part do you need help with?

**Reply by RandomMan719** [reply]
First thing to do is put the mac address of your slingbox(before turning it on), in your router and assign the slingbox an ip address based off the mac address.  Then block the slingbox from going out to the internet, so it doesn't get bricked, by sling's firmware 0.0.0.  Then you setup your slinger config.ini, using the ip address you assigned it and the port of the slingbox (differs for different models, 5001, 5201, 5301)  then turn on your slingbox, and ping it to make sure it has the correct ip address.

**Reply by sp222222** [reply]
> Yeah, I blocked them all from the internet yesterday morning. I am thinking they are bricked, won't know until I can go to each site. Hopefully we can find a way to flash the firmware, and get them functioning again.
> 
> I was able to setup a Solo (hasn't been powered up for months and I blocked it from the internet) a couple hours ago, with Slingplayer 2.0.4 and it even logged into my account and I could see all my slingboxes in the directory. I just renamed the solo, set the password, and configured a composite video source and add a custom remote for a Zenith Converter box.

must be nice  - I guess I'm a moron who cannot set router rules correctly. even though I rebooted the router once I saved all those mac addr block settings.

**Reply by GerryDazoo** [reply]
Mikey. On my router the outgoing packet blocking is done at the MAC level so you don't have to change the sling box IP address.
I'd argue that any implementation that uses IP address to block out going packets is broken. What's the purpose of blocking if it can easily be subverted by just changing your IP address.

Another point. About your box getting bricked. In the packet capture of a clean Solo factory setting reset, it was the the Setup Assistant that did the firmware download and he subsequently reflashed the the Solo. So I think that may have been the way you got bricked. SP1.5 did a firmware upgrade while you were setting up IP/password etc. So I would very heartily recommend NOT running any sling supplied software for now.

**Comment by Mike7766** [comment]
i need help with setup can you email me mike7766@gmail.com
